test/eblox/data/taxonomies/tags_test.exs

```diff
@@ -16,9 +16,16 @@ defmodule Eblox.Data.Taxonomies.TagsTest do
 
   def post_id(id), do: "#{@content_dir}/#{id}"
 
-  test "taxonomy with tags" do
-    # TODO: Wait until parsing is completed.
-    Process.sleep(100)
+  test "taxonomy with tags", context do
+    _providers_pid =
+      context.providers
+      |> GenServer.call(:which_children)
+      |> Enum.find(&match?({Eblox.Data.Providers, _, :worker, [Siblings]}, &1))
+      |> elem(1)
+
+    Siblings.transition(Eblox.Data.Providers, Eblox.Data.Provider, :update, %{pid: self()})
+    Siblings.transition(Eblox.Data.Providers, Eblox.Data.Provider, :update, %{})
+    assert_receive :on_update
```

Comment:
The first transition puts the `pid` value, and the subsequent one sends the message back, `assert_receive/1` waits for it, hence no more `Process.sleep/1` is needed.

Reply:
Wouldn't it be cleaner to provide listeners for `on_ready` event?

Something like this:

```
defmodule Eblox.Data.Provider do
  ...

  def on_transition(:init, :ready, _event_payload, payload) do
    payload
    |> Map.get(payload, :listeners, [])
    |> Enum.each(&send(&1, :on_ready))

    {:ok, :ready, payload}
  end

  ...
end
```

Reply:
It surely will.

I mixed two concepts up here. First, I [most likely properly] found it useful to provide an ability to associate some additional data with the provider during its lifecycle. Second, I [most likely mistakenly] used it also for tests.

And yes, of course, sending an event from here (or maybe even better from `on_enter(:ready, _)`) is way better.